mlflow/data/spark_dataset.py

```diff
@@ -0,0 +1,384 @@
+        """
+        base_dict.update(
+            {
+                "schema": json.dumps({"mlflow_colspec": self.schema.to_dict()}),
+                "profile": json.dumps(self.profile),
+            }
+        )
+        return base_dict
+
+    @property
+    def df(self) -> DataFrame:
+        """
+        The Spark DataFrame instance.
+
+        :return: The Spark DataFrame instance.
+
+        """
+        return self._df
+
+    @property
+    def targets(self) -> Optional[str]:
+        """
+        The name of the Spark DataFrame column containing targets (labels) for supervised
+        learning.
+
+        :return: The string name of the Spark DataFrame column containing targets.
+        """
+        return self._targets
+
+    @property
+    def source(self) -> Union[SparkDatasetSource, DeltaDatasetSource]:
+        """
+        Spark dataset source information.
+
+        :return: An instance of py:class:`SparkDatasetSource` or py:class:`DeltaDatasetSource`.
+        """
+        return self._source
+
+    @property
+    def profile(self) -> Optional[Any]:
+        # TODO: Include an approximate count of records from the table source. We don't
+        # want to compute the full count, which could be quite slow
+        return {}
+
+    @cached_property
+    def schema(self) -> Optional[Schema]:
+        """
+        The MLflow ColSpec schema of the Spark dataset.
+        """
+        try:
+            return _infer_schema(self._df)
+        except Exception as e:
+            _logger._warning("Failed to infer schema for Spark dataset. Exception: %s", e)
+            return None
+
+    def to_pyfunc(self) -> PyFuncInputsOutputs:
+        """
+        Converts the Spark DataFrame to pandas and splits the resulting
+        `pandas.DataFrame` into: 1. a `pandas.DataFrame` of features and
+        2. a `pandas.Series` of targets.
+
+        This method should only be called if the resulting `pandas.DataFrame` is expected to be
+        small, as all the data is loaded into the driver's memory.
+        """
+        df = self._df.toPandas()
+        if self._targets is not None:
+            if self._targets not in df.columns:
+                raise MlflowException(
+                    f"Failed to convert Spark dataset to pyfunc inputs and outputs because"
+                    f" the pandas representation of the Spark dataset does not contain the"
+                    f" specified targets column '{self._targets}'.",
+                    # This is an internal error because we should have validated the presence of
+                    # the target column in the Hugging Face dataset at construction time
+                    INTERNAL_ERROR,
+                )
+            inputs = df.drop(columns=self._targets)
+            outputs = df[self._targets]
+            return PyFuncInputsOutputs(inputs=inputs, outputs=outputs)
+        else:
+            return PyFuncInputsOutputs(inputs=df, outputs=None)
+
+
+def load_delta(
```

Comment:
Referring to Ray datasets, they use `read` for loading and `from` for objects. It felt intuitive to me, so it seemed reasonable to adopt a similar pattern. MLflow uses `load` for loading things from disk / network instead of `read`, so I went with `load` and `from`. Reference: https://docs.ray.io/en/latest/data/api/input_output.html#input-output.

No strong opinions here. Let me know what you think.